core/src/test/java/org/apache/iceberg/metrics/TestCounterResultParser.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.iceberg.metrics.MetricsContext.Unit;
+import org.apache.iceberg.metrics.ScanReport.CounterResult;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestCounterResultParser {
+
+  @Test
+  public void nullCounter() {
+    CounterResult<?> counterResult = CounterResultParser.fromJson((JsonNode) null);
+    Assertions.assertThat(counterResult).isEqualTo(CounterResultParser.NOOP_INT_COUNTER);
```

Comment:
I don't think this should return a result with a value. Isn't it misleading to even send noop counter results? I would omit them.

Reply:
agreed. This was from an early version where error reporting around missing fields was slightly different. I removed everything related to a noop counter/timer